src/app-sync.js

```diff
@@ -383,12 +346,31 @@ app.post('/delete-user-file', (req, res) => {
     });
   }
 
-  let rows = accountDb.all('SELECT * FROM files WHERE id = ?', [fileId]);
-
-  if (rows.length === 0) {
-    return res.status(400).send('file-not-found');
+  const filesService = new FilesService(getAccountDb());
+  if (!verifyFileExists(fileId, filesService, res, 'file-not-found')) {
+    return;
   }
 
-  accountDb.mutate('UPDATE files SET deleted = TRUE WHERE id = ?', [fileId]);
+  filesService.update(fileId, new FileUpdate({ deleted: true }));
+
   res.send(OK_RESPONSE);
 });
+
+function boolToInt(deleted) {
```

Comment:
nit: could these function definitions be moved to the top please? `verifyFileExists` and `boolToInt`